### PR TITLE
8202 llvmwinbld: Fix miscellaneous windows build issues with llvm-clang

### PR DESCRIPTION
**Pull request by jrobcary**
This should close out Issue #8202.

* Getting git date fails
* std::to{low,upp}er does not exist
* trilinos_btf_l_order and siblings use long, but the correct type on Win64 is ptrdiff_t (long long)
* Epetra_*Comm have problems because CreateDirectory is a macro in Winbase.h
* Need to define _USE_MATH_DEFINES to get M_PI defined on Windows.

An index of packages affected by commit is below.

Package    commit
@trilinos/tribits 94fe325cd4fcd6fd3de2e21454d3825bda07803a
@trilinos/epetra ed5b0c26006e87cb7ec88cf93b8a82bb0ec0af8e
@trilinos/amesos2 @trilinos/ifpack2 @trilinos/muelu 1c858848df7caecc8bbbe254cfdb5a09d40a9061
@trilinos/amesos2 c46ef3fd0bcef35f4b9fd1d993bef09f24587b5a
@trilinos/intrepid2 5b901c3a5cf1737db1959746288d3102a83241d6

## Motivation

Need to build trilinos on Windows with LLVM

## Related Issues

Will not be done until kokkos-kernels accepts PR 958 and they/we come up with a fix for Issue 967 in that project.

## Testing

All changes are in the middle of code that I presume is well tested.

## Additional Information

I am happy to modify any commit.  I tried to make my changes as local and as Windows-specific as possible, but some could be done more generally.

### CODE REVIEW

**Review by csiefer2 — APPROVED** [review]
No objection to the MueLu changes.  I'm going to approve so the autotester can run, but other packages (e.g. @trilinos/amesos2 ) should review this

File: cmake/tribits/core/package_arch/TribitsGetVersionDate.cmake
Comment (bartlettroscoe) [reply cmake/tribits/core/package_arch/TribitsGetVersionDate.cmake:38]: FYI: This is applied to TriBITS 'master' in TriBITSPub/TriBITS@ced818c.

**Comment by jwillenbring** [comment]
Since we are getting bit by random errors here, I threw the retest flag on. This is not a good long-term solution, but it is the only real option I see for the time being.

**Comment by jhux2** [comment]
@ndellingwood @srajama1 Would one of you have time to review the Amesos2 changes?

**Review by jhux2 — APPROVED** [review]
The Epetra and Intrepid2 changes look fine to me.  The @trilinos/amesos2 should approve changes there before this can be merged.

File: packages/amesos2/src/KLU2/Include/klu2_ordinaltraits.h
Comment (srajama1) [inline packages/amesos2/src/KLU2/Include/klu2_ordinaltraits.h:117]: This is somewhat off. This is a specialization for "long int". Should be just have another one for ptrdiff_t ?

**Comment by jrobcary** [comment]
This is now rebased as it had gone stale.

File: packages/amesos2/src/KLU2/Include/klu2_ordinaltraits.h
Comment (jrobcary) [reply packages/amesos2/src/KLU2/Include/klu2_ordinaltraits.h:117]: Are you asking to dispense with kluint and just use ptrdiff_t?  I can make that change if that is what you want.

Comment (jrobcary) [reply packages/amesos2/src/KLU2/Include/klu2_ordinaltraits.h:117]: I think I get it now.

**Review by jhux2 — DISMISSED** [review]
The Epetra, Ifpack2, and Intrepid2 changes look fine to me.   I'll approve, but will note that there's no need to keep the commented out code -- git will remember for us.

@srajama1 and the @trilinos/amesos2 team need to review the Amesos2 changes.

**Review by jhux2 — APPROVED** [review]
Epetra, Ifpack2, Intrepid2, MueLu all LGTM.

**Review by srajama1 — APPROVED** [review]
Looks good to me. Thanks @jrobcary !

**Comment by srajama1** [comment]
I just checked Amesos2 changes though.

**Pull request merged**
